mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -36,40 +35,34 @@
 import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
 
 import com.google.common.base.Objects;
+import com.google.common.base.Optional;
 
 public class InMemoryMailboxMapper implements MailboxMapper {
     
     private static final int INITIAL_SIZE = 128;
-    private final ConcurrentHashMap<InMemoryId, Mailbox> mailboxesById;
```

Comment:
Id unicity is due to Atomic Long use.

To ensure atomicity and unicity of path, we need to serve as a basis for the  concurrent structure. Otherwize concurrency bugs hangs around.